scalafix-core/src/main/scala/scalafix/internal/patch/EscapeHatch.scala

```diff
@@ -132,17 +125,23 @@ object EscapeHatch {
       id.identifiers.exists(i => matcher.matches(i.value))
   }
 
-  private case class EscapeOffset(offset: Int)
+  private case class EscapeOffset(offset: Int) extends AnyVal
 
   private object EscapeOffset {
     implicit val ordering: Ordering[EscapeOffset] = Ordering.by(_.offset)
   }
 
-  def apply(tree: Tree, associatedComments: AssociatedComments): EscapeHatch =
+  def apply(
+      input: Input,
+      tree: => Tree,
```

Comment:
Note: my experience with by-name is that it's easy to accidentally trigger the value. You can also use `() => Tree` or the `LazyValue[T]` in https://github.com/scalacenter/scalafix/blob/18650f572178ab1fa461846604bd82f9fed996ac/scalafix-core/src/main/scala/scalafix/internal/v1/LazyValue.scala which has the benefit of being a normal type and avoids accidental double computations. I prefer explicit `tree.value` or `tree()` to make it clearer when the function is called.

Reply:
done